SDBlockDevice.cpp

```diff
@@ -363,6 +364,12 @@ int SDBlockDevice::_initialise_card()
 
 int SDBlockDevice::init()
 {
+    uint32_t val = core_util_atomic_incr_u32(&_init_ref_count, 1);
```

Comment:
Increment on first try of init will increase the count and retries in case of failure will be skipped.

Either move this increment after init is successful, or else decrement/reset value  in case of init failure.
https://github.com/ARMmbed/sd-driver/pull/107/commits/5320df33b1ad7e15a59b218445447e1229720559#diff-9bbd60f38e5eac7ba763b00ec6d6d447R378

Reply:
Good point. Are retries relevant here? I mean, do you see a case where an initialization fails at first try, but succeeds in the next one?

Reply:
Yes I believe cloud client and other applications do perform retries